app/views/handlers/question.py

```diff
@@ -192,6 +192,9 @@ def handle_post(self):
         self.questionnaire_store_updater.update_answers(self.form.data)
         self.questionnaire_store_updater.update_progress_for_dependent_sections()
         if self.questionnaire_store_updater.is_dirty():
+            # We prematurely complete the block, as we need it completed to build the routing path
+            # In order to support progress value source references of the previous block
+            self.questionnaire_store_updater.add_completed_location()
```

Comment:
The schema `test_progress_source_repeating_sections.json` doesn't work for the unhappy path.
- In section A, add an item and when you get to random question enabler, go to the Hub
- Complete the section for item A
- Go to section A and now answer the random question enabler
- Return to the Hub and see that Item A section says Completed, but it should say in progress because a new question was opened up.

The issue is because `update_progress_for_dependent_sections` is called before `add_completed_location`.
A new test will be needed for this.

Reply:
Fixed and integration test amended accordingly